src/aws_encryption_sdk/materials_managers/mpl/materials.py

```diff
@@ -0,0 +1,138 @@
+# Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+"""Provides encryption/decryption materials from an underlying materials provider from the MPL.
+
+The aws-cryptographic-materials-library MUST be installed to use this module.
+"""
+# pylint should pass even if the MPL isn't installed
+# noqa pylint: disable=import-error
+from aws_cryptographic_materialproviders.mpl.models import (
+    DecryptionMaterials as MPL_DecryptionMaterials,
+    EncryptedDataKey as MPL_EncryptedDataKey,
+    EncryptionMaterials as MPL_EncryptionMaterials,
+)
+# pylint and isort disagree on where this should go. Choose isort and disable pylint for this.
+from typing import Dict, List, Set  # noqa pylint: disable=wrong-import-order
+
+from aws_encryption_sdk.identifiers import Algorithm, AlgorithmSuite
+from aws_encryption_sdk.materials_managers import (
+    DecryptionMaterials as Native_DecryptionMaterials,
+    EncryptionMaterials as Native_EncryptionMaterials,
+)
+from aws_encryption_sdk.structures import DataKey, EncryptedDataKey as Native_EncryptedDataKey, MasterKeyInfo
+
+
+def _mpl_algorithm_id_to_native_algorithm_id(mpl_algorithm_id: str) -> int:
+    # MPL algorithm suite ID == hex(native algorithm suite ID)
+    return int(mpl_algorithm_id, 16)
+
+
+class EncryptionMaterialsFromMPL(Native_EncryptionMaterials):
```

Comment:
What about required EC and the like?

Reply:
I'll plan to leave all required EC "stuff" in https://github.com/aws/aws-encryption-sdk-python/pull/645